enhancements/user-monitoring.md

```diff
@@ -0,0 +1,248 @@
+
+#### Isolation
+
+The cluster monitoring as well as the user monitoring stack are composed of custom resources
+reconciled by two separate prometheus operators.
+Both prometheus operators are under supervision of cluster monitoring operator.
+This allows a seperation of concerns in terms of ugpradability, security settings, and scalability.
+The cluster monitoring Prometheus Operator continues to be deployed in the `openshift-monitoring` namespace.
+The user monitoring Prometheus Operator will be deployed in the `openshift-user-monitoring` namespace.
+
+Currently, cluster monitoring operator scans all namespaces filtering openshift namespaces labelled with
+`openshift.io/cluster-monitoring: "true"` and takes this set of namespaces as the _allowed list_ of reconcilable
+namespaces for the existing cluster monitoring prometheus operator.
+
+The same set of openshift namespaces that exist today will be configured as a _deny list_ of reconcilable namespaces
+at the user monitoring prometheus operator.
+Deny listing of namespaces was added recently in prometheus operator, see [3].
+
+To make sure that users cannot add Prometheus instances in user namespaces,
+a new WIP feature [4] will ensure that reconcilation of Prometheus custom resources
+will be limited to the `openshift-user-monitoring` namespace for the user monitoring prometheus operator
+and `openshift-monitoring` namespace for the cluster monitoring prometheus operator.
+
+![](user-namespaces-custom-resources.png)
+
+User monitoring Prometheus Operator (green) then will reconcile Prometheus custom resources
+in the `openshift-user-monitoring` namespace only.
+It ignores Prometheus custom resources in any other namespace.
+The same holds true for the cluster monitoring prometheus operator (red).
+
+#### Aggregation
+
+The front facing Kubernetes service has to make it as simple as possible
+for integrators to leverage both user monitoring as well as cluster monitoring.
+
+This is achieved by using Thanos Querier which allows to aggregate
+and optionally deduplicate the cluster and user monitoring metrics backends
+under single Prometheus Query endpoint.
+
+Usage of the Thanos querier component inside cluster monitoring has many advantages:
+1. Run-time deduplication of metrics available today.
+2. Potentially gap-filling of metrics series in the Prometheus HA user and cluster monitoring setup in the future.
+3. Potentially add the possibility of adding a query cache for increased performance in the future.
+
+See [5] and [6] for more details.
+
+### Tenancy
+
+![](user-monitoring-request.png)
+
+Tenancy is achieved by leveraging the existing topology that is protecting cluster monitoring Prometheus already today.
+The kube-rbac-proxy sidecar will be deployed along with prom-label-proxy in the Thanos Querier Deployment.
+
+### Risks and Mitigations
+
+Risks are mainly in the area of security, namely in the realm of accessing metrics a user has no permissions for.
```

Comment:
Well I would say risks are also:

* QoD (we mitigate some of it by timeouts, but the saturation of services can be done) 
* Prometheus unavailability by ingestion due to aggressive scrape target with high cardinality

Reply:
spelled out QoD = Query of Death, and added those bullet points too :+1: